src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Binders/EENamedTypeBinder.vb

```diff
@@ -59,7 +59,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
                 For i As Integer = 0 To symbols.Count - 1
                     Dim symbol = symbols(i)
                     If symbol.Kind = SymbolKind.TypeParameter Then
-                        Debug.Assert(symbol.OriginalDefinition.ContainingSymbol = substitutedSourceType.OriginalDefinition)
+                        Debug.Assert(TypeSymbol.Equals(DirectCast(symbol.OriginalDefinition.ContainingSymbol, TypeSymbol), substitutedSourceType.OriginalDefinition, TypeCompareKind.ConsiderEverything))
```

Comment:
Yes, it should always be safe to cast because `substitutedSourceType` is a type symbol, so its original definition should be a type symbol too, and the previous assertion means that `symbol.OriginalDefinition.ContainingSymbol` matches that.

---
In reply to: [250745961](https://github.com/dotnet/roslyn/pull/31943#discussion_r250745961) [](ancestors = 250745961)